Dockerfile

```diff
@@ -1,4 +1,4 @@
-ARG GO_VERSION=1.16
+ARG GO_VERSION=1.17
```

Comment:
Given that 1.16 is still supported, I'm slightly leaning towards sticking to that version (and not update the default for a patch release)

Reply:
CI only uses `Dockerfile` for making release binaries (see `release` job in `.github/workflows/validate.yml`), and I think using latest Go for building binaries is slightly better.

Also, I'm not sure whether @cyphar will build a release via Dockerfile or not, but in case he will be using Dockerfile, sticking to latest Go for release binaries is sort of a must.

Reply:
I think there's risk involved in switching go runtimes. We've hit odd regressions on various Golang updates in Moby / Docker Engine, so to reduce risk I'd rather do this for `v1.1.x`, unless we have a strong reason to move to 1.17

Reply:
I don't use Docker to build the releases so I use the latest Go release on my openSUSE Tumbleweed machine, though for the cross-builds I have to use Docker. At some point (once I sit down and finally hammer out the last bits of the `release.sh` improvements) I will probably use Docker for everything.

However, I agree with @thaJeztah. I remember being burned by Go releases many times in the past, and while things do appear more stable these days, Go 1.14 broke many Go programs thanks to their preemptive scheduling changes. (Though in practice distributions will build the binaries with whatever Go version they want -- in openSUSE we track upstream and only build with the Go version used by upstream but it's possible other distributions don't.)

Reply:
I do remember my share of woes switching to higher Go versions, both in moby/moby and here, but the last big one was EINTR and it's a few releases back. IOW things are getting more stable, or so it seems.

Nevertheless this is a stable branch, so I agree in this case it's better to not change horses in the middle of the stream.

Reverted.